### PR TITLE
fix(status): do not set Unknown back to component for every loop on reconcile

### PR DESCRIPTION
**Pull request by zdtsw**
ref: https://issues.redhat.com/browse/RHOAIENG-415
## How Has This Been Tested?

local build quay.io/wenzhou/opendatahub-operator-catalog:v2.10.415-1

## Merge criteria:

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: controllers/datasciencecluster/datasciencecluster_controller.go [hunk 1]
```diff
@@ -278,13 +278,21 @@ func (r *DataScienceClusterReconciler) reconcileSubComponent(ctx context.Context
 	componentName := component.GetComponentName()
 
 	enabled := component.GetManagementState() == v1.Managed
+
+	isExistStatus := false
```
Comment (ykaliuta) [inline controllers/datasciencecluster/datasciencecluster_controller.go:282]: Probably no need, you have assignment right below.

File: controllers/datasciencecluster/datasciencecluster_controller.go [hunk 2]
```diff
@@ -278,13 +278,21 @@ func (r *DataScienceClusterReconciler) reconcileSubComponent(ctx context.Context
...
 	// First set conditions to reflect a component is about to be reconciled
 	instance, err := status.UpdateWithRetry(ctx, r.Client, instance, func(saved *dsc.DataScienceCluster) {
 		message := "Component is disabled"
 		if enabled {
 			message = "Component is enabled"
 		}
-		status.SetComponentCondition(&saved.Status.Conditions, componentName, status.ReconcileInit, message, corev1.ConditionUnknown)
+
+		// only set to init condition e.g Unknonw for the very first time when component is not in the list
+		if !isExistStatus {
```
Comment (ykaliuta) [inline controllers/datasciencecluster/datasciencecluster_controller.go:293]: Looks like the whole update is under the condition

Comment (zdtsw) [reply controllers/datasciencecluster/datasciencecluster_controller.go:291]: yes, the idea is to, only make this call, if we find the component in the installcomponents list,
it indicates if this is happening after upgrade operator or it is another new reconcile. so we can skip the Unknown showing after it is True or False

Comment (ykaliuta) [reply controllers/datasciencecluster/datasciencecluster_controller.go:291]: Yeah, but should you skip UpdateWithRetry() call in this case at all?

Comment (zdtsw) [reply controllers/datasciencecluster/datasciencecluster_controller.go:291]: ah, i see your point.

**Review by ykaliuta — APPROVED** [review]

**Comment by zdtsw** [comment]
the failure in the e2e is due to a missing tag on CFO 1.4.0
i will merge this PR now.

**Pull request merged**